src/Compilers/Core/Portable/SourceGenerator/SourceGeneratorExtensions.cs

```diff
@@ -16,7 +16,9 @@ public static class SourceGeneratorExtensions
             ImmutableArray<SourceGenerator> generators,
             string path,
             bool writeToDisk,
-            CancellationToken cancellationToken)
+            Func<Exception, bool> exceptionFilter,
```

Comment:
Is the only purpose of the exception filter for testing? It feels like a strange amount of power to give the caller.